LanguageFeatures/primary-constructors/static_processing_A05_t01.dart

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A compile-time error occurs if the name of the primary
+/// constructor is the same as the name of a constructor (declaring or not) in
+/// the body.
+///
+/// @description Check that it is a compile-time error if the name of the
+/// primary constructor is the same as the name of some constructor declared in
+/// the body. Test classes.
+/// @author sgrekhov22@gmail.com
+
+// TODO (sgrekhov) Add `declaring-constructors` experimental flag
+
+class C1(var int v) {
+  C1.someName(this.v);
+  C1(int v) : this.someName(v);
+//^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class C2(int v1) {
+  this(var int v2);
+//^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class const C3.someName(final int v) {
+  const C3(this.v);
+  const C3.someName(int v) : this(v);
+//      ^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+}
+
+class C4.someName(int v1) {
+  this.someName(var int v2);
```

Comment:
It's the same confusing situation as some time earlier: This could be reported as an error because there are two declaring formal parameter lists in the class (including the header). In this case the in-header constructor and the in-body constructor are the same constructor, and it's an error to have `(var int v2)` in the body.

 It could also be reported as an error because the two declarations are two constructors, and then they can't have the same name.